sdk/nodejs/automation/stack.ts

```diff
@@ -587,6 +587,47 @@ Event: ${line}\n${e.toString()}`);
         };
     }
 
+    /**
+     * Rename an existing stack
+     */
+    async rename(options: RenameOptions): Promise<RenameResult> {
+        const args = ["stack", "rename", options.stackName];
+        args.push(...this.remoteArgs());
+
+        applyGlobalOpts(options, args);
+
+        let logPromise: Promise<ReadlineResult> | undefined;
+        let logFile: string | undefined;
+
+        // Set up event log tailing
+        if (options?.onEvent) {
+            const onEvent = options.onEvent;
+            logFile = createLogFile("rename");
+            args.push("--event-log", logFile);
+
+            logPromise = this.readLines(logFile, (event) => {
+                onEvent(event);
+            });
+        }
+
+        let renameResult: CommandResult;
+        try {
+            renameResult = await this.runPulumiCmd(args, options?.onOutput, options?.signal);
+        } finally {
+            await cleanUp(logFile, await logPromise);
+        }
+
+        // If it's a remote workspace, explicitly set showSecrets to false to prevent attempting to
+        // load the project file.
+        const summary = await this.info(!this.isRemote && options?.showSecrets);
```

Comment:
As with python, I wonder if we should error out if the user explicitly sets `showSecrets` and this is a remote workspace.